### PR TITLE
chore: Only run CI checks when necessary (affected files changed)

### PR DESCRIPTION
**Pull request by bryantbiggs**
### What does this PR do?
- Only run CI checks when necessary (affected files changed)

### Motivation
- Reduce the amount of checks that get executed that are not necessary for the given changeset

### More

- [ ] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Review by Zvikan — APPROVED** [review]

File: .github/workflows/plan-examples.yml [hunk 1]
```diff
@@ -42,21 +50,34 @@ jobs:
...
-      - name: checkout-merge
-        if: "contains(github.event_name, 'pull_request')"
-        uses: actions/checkout@v3
-        with:
-          ref: refs/pull/${{github.event.pull_request.number}}/merge
+      # - name: checkout-merge
+      #   if: "contains(github.event_name, 'pull_request')"
+      #   uses: actions/checkout@v3
+      #   with:
+      #     ref: refs/pull/${{github.event.pull_request.number}}/merge
```
Comment (bryantbiggs) [inline .github/workflows/plan-examples.yml:60]: @Zvikan do you know why we need the two different routes for checking out? plans only run on PRs so the standard checkout should suffice, no?

**Comment by bryantbiggs** [comment]
waiting till issue is resolved, explains some of the "weirdness" in the checks that have been occurring 


image

File: .github/workflows/plan-examples.yml (hunk 1 above)
Comment (Zvikan) [reply .github/workflows/plan-examples.yml]: According to https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/572
"Explicit checkout PR code as pull_request_target changes checkout behavior to default to source main."

Comment (bryantbiggs) [reply .github/workflows/plan-examples.yml]: ah ok thanks - I put that back as it was

**Pull request merged**